board.h

```diff
@@ -0,0 +1,61 @@
+#include <vector>
+#include <iostream>
+#include <string>
+#include <cstdlib> 
+
+struct Position{
+    unsigned int xpos = 0;
+    unsigned int ypos = 0;
+
+    bool operator==(const Position &b) const;
+};
+
+struct Mine:Position{
+    int owner = -1;
+
+    bool operator==(const Mine &b) const;
+
+    bool operator==(const Position &b) const;
+};
+
+struct Player{
+    int id = -1;
+    Mine* mines = nullptr;
+    int mineCount = 0;
+    bool operator==(const Player b){
+        return ((id == b.id));
+    }
+    bool isAI = false;
+};
+
+enum gameType{

```

Comment:
#should use CamelCase instead of dromedaryCase.

#should use `enum class` instead.